common/data_refinery_common/models/models.py

```diff
@@ -737,9 +743,10 @@ def needs_processing(self, own_processor_id=None) -> bool:
         if own_processor_id:
             incomplete_processor_jobs = incomplete_processor_jobs.exclude(id=own_processor_id)
 
-        # If the file has a processor job that hasn't even started yet,
-        # then it doesn't need another.
-        if incomplete_processor_jobs.count() > 0:
+        # Check if there's any jobs which should block another
+        # processing attempt.
+        blocking_jobs = no_retry_processor_jobs | incomplete_processor_jobs
```

Comment:
cool, didn't know you could combine querysets like this.

Reply:
Yeah me neither! You can use the `&` operator too apparently!